src/identity-utils.ts

```diff
@@ -17,74 +30,101 @@ export type IParseCachedIdentityResponse = (
     identityMethod: string,
     knownIdentities: IKnownIdentities,
     fromCachedIdentity: boolean
-) => void
+) => void;
 
 export interface IKnownIdentities extends UserIdentities {
     device_application_stamp?: string;
 }
 
 export interface ICachedIdentityCall {
+    // https://go.mparticle.com/work/SQDSDKS-6672
     responseText: string;
     status: number;
     expireTimestamp: number;
 }
 
+// https://go.mparticle.com/work/SQDSDKS-6568
+// Temporary adapter to convert the XMLHttpRequest response to the IIdentityResponse interface
+export const xhrIdentityResponseAdapter = (
+    possiblyXhr: XMLHttpRequest | IIdentityResponse
+): IIdentityResponse => {
+    if (possiblyXhr.hasOwnProperty('expireTimestamp')) {
+        // This is likely an IIdentityResponse object, just return it
```

Comment:
It is not clear that the `IIdentityResponse` interface is cached.

Also - when you say `likely`, that implies it isn't the case 100% of the time, but if it goes down this code path, then it in fact *is* `IIdentityResponse`, so I think the comment should be explicit instead of saying `likely`.
```suggestion
        // If there is an `expireTimestamp`, it is an IIdentityResponse object, so just return it.  This indicates it was a previously cached value.
```